### ISSUE TITLE
Deleting data source with Advanced Scripting

### ISSUE THREAD

**Issue by m-kovalsky** [otykier/TabularEditor#824]
**Describe the bug**
When using Advanced Scripting to delete a data source, it does not provide an error message as the UI does when at least 1 partition uses that data source.

**To Reproduce**

Make sure that the data source used below is referenced by at least one table/partition.

1. Run this in the Advanced Scripting window 

string dsName = "SqlServer ServerName";
Model.DataSources[dsName].Delete();

2. No error appears
3. If you right click on the data source and press Delete, you get the following error:

![image](https://user-images.githubusercontent.com/29556918/114345036-a2322a00-9b69-11eb-9589-6bace5c554d8.png)

**Expected behavior**
I believe the same error should show whether deleting via the UI (right click -> delete) or using Advanced Scripting.

**Application specifics**
 - Tabular Editor version: 2.15.3

**Comment by otykier** [comment]
This is actually by design. The UI performs a few additional checks in various places to prevent setting TOM properties in a way that would cause errors when deploying/saving to AS (granted, there are many places where the UI could do better in this area). Script authors are assumed to have more in-depth TOM knowledge, and there may very well be situations where you'd like a script to perform some actions in an order that would normally not be possible in the UI.

In other words, scripting lets you work more directly with the TOM, with all the pros and cons that this entails.

**Comment by m-kovalsky** [comment]
Got it, thanks for the explanation.
